manager/allocator/networkallocator/portallocator.go

```diff
@@ -91,40 +139,63 @@ func reconcilePortConfigs(s *api.Service) []*api.PortConfig {
 		return s.Spec.Endpoint.Ports
 	}
 
-	allocatedPorts := make(map[api.PortConfig]*api.PortConfig)
+	portStates := allocatedPorts{}
 	for _, portState := range s.Endpoint.Ports {
 		if portState.PublishMode != api.PublishModeIngress {
 			continue
 		}
-
-		allocatedPorts[getPortConfigKey(portState)] = portState
+		portStates.addState(portState)
 	}
 
 	var portConfigs []*api.PortConfig
+
+	// Process the portConfig with portConfig.PublishMode != api.PublishModeIngress
+	// Iterate portConfigs with PublishModeIngress
 	for _, portConfig := range s.Spec.Endpoint.Ports {
 		// If the PublishMode is not Ingress simply pick up
 		// the port config.
 		if portConfig.PublishMode != api.PublishModeIngress {
 			portConfigs = append(portConfigs, portConfig)
 			continue
 		}
+	}
+
+	// Iterate portConfigs with PublishedPort != 0 (high priority)
+	for _, portConfig := range s.Spec.Endpoint.Ports {
+		// Ignore ports which are not PublishModeIngress (already processed)
+		if portConfig.PublishMode != api.PublishModeIngress {
+			continue
+		}
```

Comment:
Maybe this loop should iterate over `portConfigs` so it doesn't need to check the `PublishMode`.